build-tools/packages/build-cli/docs/release.md

```diff
@@ -52,55 +53,99 @@ Generates a report of Fluid Framework releases.
+      repo, but will also include its direct Fluid dependencies.
+
+      If you want to report on a particular release, check out the git tag for the release version you want to report on
+      before running this command.
+      <options: client|server|azure|build-tools>
+
+  -i, --interactive
+      Choose the version of each release group and package to contain in the release report.
+
+  -o, --[no-]output
+      Output JSON report files.
+
+  -p, --path=<value>
+      [default: working directory] Output JSON report files to this location.
+
+  -r, --mostRecent
+      Always pick the most recent version as the latest (ignore semver version sorting).
+
+  -s, --highest
+      Always pick the greatest semver version as the latest (ignore dates).
+
+  -v, --verbose
+      Verbose logging.
 
 GLOBAL FLAGS
   --json  Format output as json.
 
 DESCRIPTION
   Generates a report of Fluid Framework releases.
 
-  The release report command is used to produce a report of all the packages that were released and their current
-  version. After a release, it is useful to generate this report to provide to customers, so they can update their
-  dependencies to the most recent version.
+  The release report command is used to produce a report of all the packages that were released and their version. After
+  a release, it is useful to generate this report to provide to customers, so they can update their dependencies to the
+  most recent version.
 
-  The command will prompt you to select versions for a package or release group in the event that multiple versions have
-  recently been released.
+  The command operates in two modes: "whole repo" or "release group." The default mode is "whole repo." In this mode,
+  the command will look at the git tags in the repo to determine the versions, and will include all release groups and
+  packages in the repo. You can control which version of each package and release group is included in the report using
+  the --interactive, --mostRecent, and --highest flags.
 
-  Using the --all flag, you can list all the releases for a given release group or package.
+  The "release group" mode can be activated by passing a --releaseGroup flag. In this mode, the specified release
+  group's version will be loaded from the repo, and its immediate Fluid dependencies will be included in the report.
+  This is useful when we want to include only the dependency versions that the release group depends on in the report.
 
 EXAMPLES
   Output all release report files to the current directory.
 
-    $ flub release report -o .
+    $ flub release report
 
-  Generate a minimal release report and display it in the terminal.
+  Generate a release report of the highest semver release for each package and release group and display it in the
+  terminal only.
 
-    $ flub release report
+    $ flub release report --no-output
```

Comment:
I don't think "--no-output" implies that the output goes to the terminal. Its also unclear if the default is to write the json files or not. Personally, I'd interpret "--no-output" like "--quite" and expect it to write the files, but suppress console output.

Having a flag which lets you pick where to save the files, and if not provided, not creating (or writing them to the terminal) seems like a more common approach.

Reply:
> Having a flag which lets you pick where to save the files, and if not provided, not creating (or writing them to the terminal) seems like a more common approach.

That's how it worked before. I changed it because I thought that outputting the files by default was better. But I agree it's unusual. I'll switch it back. `-o .` will output the files to the working directory; omitting it will output to the terminal only.